Lib/string.py

```diff
@@ -64,14 +70,24 @@ class Template:
     # See https://bugs.python.org/issue31672
     idpattern = r'(?a:[_a-z][_a-z0-9]*)'
     braceidpattern = None
-    flags = _re.IGNORECASE
+    flags = None  # default: re.IGNORECASE
+
+    pattern = _TemplatePattern()  # use a descriptor to compile the pattern
 
     def __init_subclass__(cls):
         super().__init_subclass__()
-        if 'pattern' in cls.__dict__:
-            pattern = cls.pattern
+        cls._compile_pattern()
+
+    @classmethod
+    def _compile_pattern(cls):
+        import re  # deferred import, for performance
+
+        cls_pattern = cls.__dict__.get('pattern')
+        if cls_pattern and not isinstance(cls_pattern, _TemplatePattern):
```

Comment:
Could not the `_TemplatePattern` instance be a singleton? Then this check could be simpler.

Reply:
> Is `__init_subclass__()` needed anymore?

I think so, as if a subclass is created after `Template.pattern` has been replaced by the compiled pattern, the subclass will just inherit that attribute, rather than freshly compiling the regex for that subclass.

> Could not the `_TemplatePattern` instance be a singleton? Then this check could be simpler.

I'm not sure I follow, how would `_TemplatePattern` be made into a singleton?

Reply:
```py
_TemplatePattern = _TemplatePattern()
```
Then use `is` instead of `isinstace()`.